miner/proposal_builder_test.go

```diff
@@ -265,33 +265,33 @@ func TestBuild_BlockedSignerInitDoesntBlockEligible(t *testing.T) {
 		syncer    = smocks.NewMockSyncStateProvider(ctrl)
 		db        = statesql.InMemoryTest(t)
 		localdb   = localsql.InMemoryTest(t)
-		atxsdata  = atxsdata.New()
+		data      = atxsdata.New()
 		// singer[1] is blocked
-		atxSearch = mocks.NewMockatxSearch(ctrl)
+		dataMock = mocks.NewMockatxsData(ctrl)
 	)
 	opts := []Opt{
 		WithLayerPerEpoch(types.GetLayersPerEpoch()),
 		WithLayerSize(2),
 		WithLogger(zaptest.NewLogger(t)),
 		WithSigners(signers...),
-		withAtxSearch(atxSearch),
 	}
-	builder := New(clock, db, localdb, atxsdata, publisher, trtl, syncer, conState, opts...)
+	builder := New(clock, db, localdb, dataMock, publisher, trtl, syncer, conState, opts...)
 	lid := types.LayerID(15)
 
 	// only signer[0] has ATX
 	atx := gatx(types.ATXID{1}, lid.GetEpoch()-1, signers[0].NodeID(), 1, genAtxWithNonce(777))
 	require.NoError(t, atxs.Add(db, atx, types.AtxBlob{}))
-	atxsdata.AddFromAtx(atx, false)
-	atxSearch.EXPECT().GetIDByEpochAndNodeID(gomock.Any(), lid.GetEpoch()-1, signers[0].NodeID()).DoAndReturn(
-		func(_ context.Context, epoch types.EpochID, nodeID types.NodeID) (types.ATXID, error) {
-			return atxs.GetIDByEpochAndNodeID(db, epoch, nodeID)
+	data.AddFromAtx(atx, false)
+	dataMock.EXPECT().GetByEpochAndNodeID(lid.GetEpoch(), signers[0].NodeID()).DoAndReturn(
+		func(epoch types.EpochID, nodeID types.NodeID) (types.ATXID, *atxsdata.ATX) {
+			return data.GetByEpochAndNodeID(epoch, nodeID)
 		},
 	)
```

Comment:
Two NITs:
1. the ATX doesn't need to go to the DB anymore (`require.NoError(t, atxs.Add(db, atx, types.AtxBlob{}))` could be removed).
2. The `data` isn't needed as well, there is `dataMock`, could just return the generated (atxid, ATX) tuple

Reply:
I simplified the test to only use the `dataMock` and not a real implementation of the cache, but I had to keep the ATX in the DB. Otherwise the test fails because no active set can be generated (no ATXs in DB)